packages/peregrine/lib/talons/CartPage/GiftCards/useGiftCards.js

```diff
@@ -72,46 +58,25 @@ export const useGiftCards = props => {
     const [applyCard, applyCardResult] = useMutation(applyCardMutation);
     const [removeCard, removeCardResult] = useMutation(removeCardMutation);
 
-    /*
-     * useMemo hooks / member variables.
-     */
-    const numCards = useMemo(() => {
-        return cartResult.data
-            ? cartResult.data.cart.applied_gift_cards.length
-            : Number.NEGATIVE_INFINITY;
-    }, [cartResult.data]);
-
-    const canTogglePromptState = numCards > 0;
-    const initialPromptState = getPromptStateForNumCards(numCards);
-
     /*
      *  useState hooks.
      */
     const [formApi, setFormApi] = useState();
     const [mostRecentAction, setMostRecentAction] = useState(null);
-    const [promptState, setPromptState] = useState(initialPromptState);
 
     /*
      *  useEffect hooks.
      */
     // Fire the getCartDetails query immediately and whenever cartId changes.
     useEffect(() => {
-        if (!cartId) {
-            return;
+        if (cartId) {
+            getCartDetails({
```

Comment:
This is actually getting applied gift cards, consider a rename while we're here.

Reply:
Done in b87b2e757